pkg/webhook/elasticquota/quota_topology_check.go

```diff
@@ -0,0 +1,259 @@
+		if isLessEqual, _ := quotav1.LessThanOrEqual(childMinSumIncludeSelf, qt.quotaInfoMap[quotaInfo.ParentName].CalculateInfo.Min); !isLessEqual {
+			return fmt.Errorf("checkMinQuotaSum allChildren SumMinQuota > parentMinQuota, parent: %v", quotaInfo.ParentName)
+		}
+	}
+
+	children, exist := qt.quotaHierarchyInfo[quotaInfo.Name]
+	if !exist || len(children) == 0 {
+		return nil
+	}
+
+	childMinSum, err := qt.getChildMinQuotaSumExceptSpecificChild(quotaInfo.Name, "")
+	if err != nil {
+		return fmt.Errorf("checkMinQuotaSum failed:%v", err)
+	}
+
+	if isLessEqual, _ := quotav1.LessThanOrEqual(childMinSum, quotaInfo.CalculateInfo.Min); !isLessEqual {
+		return fmt.Errorf("checkMinQuotaSum allChildrn SumMinQuota > MinQuota, parent: %v", quotaInfo.Name)
+	}
+
+	return nil
+}
+
+func (qt *quotaTopology) getChildMinQuotaSumExceptSpecificChild(parentName, skipQuota string) (allChildQuotaSum v1.ResourceList, err error) {
+	allChildQuotaSum = v1.ResourceList{}
+	if parentName == extension.RootQuotaName {
+		return allChildQuotaSum, nil
+	}
+
+	childQuotaList, exist := qt.quotaHierarchyInfo[parentName]
+	if !exist {
+		return nil, fmt.Errorf("not found child quota list, parent: %v", parentName)
+	}
+
+	for childName := range childQuotaList {
+		if childName == skipQuota {
+			continue
+		}
+
+		if quotaInfo, exist := qt.quotaInfoMap[childName]; exist {
+			allChildQuotaSum = quotav1.Add(allChildQuotaSum, quotaInfo.CalculateInfo.Min)
+		} else {
+			err = fmt.Errorf("BUG quotaInfoMap and quotaTree information out of sync, losed :%v", childName)
+			return nil, err
+		}
+	}
+
+	return allChildQuotaSum, nil
+}
+
+func toElasticQuota(obj interface{}) *v1alpha1.ElasticQuota {
+	if obj == nil {
+		return nil
+	}
+
+	var unstructuredObj *unstructured.Unstructured
+	switch t := obj.(type) {
+	case *v1alpha1.ElasticQuota:
+		return obj.(*v1alpha1.ElasticQuota)
+	case *unstructured.Unstructured:
+		unstructuredObj = obj.(*unstructured.Unstructured)
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		unstructuredObj, ok = t.Obj.(*unstructured.Unstructured)
+		if !ok {
+			klog.Errorf("Fail to convert quota object %T to *unstructured.Unstructured", obj)
+			return nil
+		}
+	default:
+		klog.Errorf("Unable to handle quota object in %T", obj)
+		return nil
+	}
+
+	quota := &v1alpha1.ElasticQuota{}
+	err := scheme.Scheme.Convert(unstructuredObj, quota, nil)
+	if err != nil {
+		klog.Errorf("Fail to convert unstructed object %v to Quota: %v", obj, err)
+		return nil
+	}
+	return quota
+}
+
+func quotaFieldsCopy(q *v1alpha1.ElasticQuota) v1alpha1.ElasticQuota {
+	return v1alpha1.ElasticQuota{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				extension.LabelQuotaParent:       q.Labels[extension.LabelQuotaParent],
+				extension.LabelQuotaIsParent:     q.Labels[extension.LabelQuotaIsParent],
+				extension.LabelAllowLentResource: q.Labels[extension.LabelAllowLentResource],
```

Comment:
LabelAllowLentResource should be killed cause you don't check it.  and you forgot "Min"